portfolio/src/main/webapp/script.js

```diff
@@ -15,11 +15,16 @@
 /** onload function that displays comments and comment form/login information */
 function displayCommentSection() {
     getComments();
```

Comment:
This is a bit tricky and can cause a race condition. getComments() is async code, so it's possible that everything below will run before the code in getComments() has updated the DOM.

You can re-factor this a bit by making getComments a function that returns the promise chain (instead of being void). You can then add it to your promise chain here. e.g:

getComments().then( () => fetch('/login') ).then(  .....

Reply:
I fixed this, not exactly how you suggested but hopefully in a way that accomplishes the same thing.

Reply:
Yep, that way works too, as long as it's okay that displayCommentSection() will return before getComments() is done executing. Seems like that's not a problem though.

Reply:
yeah, neither are dependent on each other